sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala

```diff
@@ -91,6 +93,16 @@ object SQLMetrics {
 
   private val baseForAvgMetric: Int = 10
 
+  val cachedSQLAccumIdentifier = Some(AccumulatorContext.SQL_ACCUM_IDENTIFIER)
+
+  private val metricsCache: LoadingCache[String, Option[String]] =
+    CacheBuilder.newBuilder().maximumSize(10000)
```

Comment:
I guess `10000` is fine. I wonder if it's better to reduce the number to like 128 though.

Reply:
Do you think we wouldn't need to cache more than the top 128 metric names. It does makes sense. Just to be on the safer side, I set it to 10k. I don't think this should consume a lot of memory though. I am fine either way. Let me know your thoughts.